### PR TITLE
fix: add basic validation in otlphttpexporter

### LINKED ISSUE #4709: No validations in otlphttpexporter
**Describe the bug**

Validate method does not evaluate the given config for the exporter.
https://github.com/open-telemetry/opentelemetry-collector/blob/04baaae533dc2fea3bfddce1df1c72c3d0694042/exporter/otlphttpexporter/config.go#L42

**Steps to reproduce**
If possible, provide a recipe for reproducing the error.

When collector is started with no value for endpoint in config.yaml for otlphttpexporter, collector crashes saying either endpoint or traces_endpoint (for traces pipeline) must be defined. Similar behaviour is observed for metrics and logs pipeline.

**What did you expect to see?**
A clear and concise description of what you expected to see.

These checks should be performed in Validate method, instead they are being done in factory.go in composeSignalURL method.
https://github.com/open-telemetry/opentelemetry-collector/blob/04baaae533dc2fea3bfddce1df1c72c3d0694042/exporter/otlphttpexporter/factory.go#L60

**What did you see instead?**
A clear and concise description of what you saw instead.

**What version did you use?**
Version: (e.g., `v0.4.0`, `1eb551b`, etc)

**What config did you use?**
Config: (e.g. the yaml config file)

image

**Environment**
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")

**Additional context**

### PR DESCRIPTION
**Pull request by anupamdalmia10**
**Description:** <Describe what has changed. 
Adding basic validation to check if at least one endpoint is specified in otlphttpexporter configuration

**Link to tracking Issue:** <Issue number if applicable>
Closes #4709 

**Testing:** < Describe what testing was performed and which tests were added.>
Tested with no endpoints specified and at least one endpoint specified.

### CODE REVIEW

File: exporter/otlphttpexporter/config_test.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ func TestLoadConfig(t *testing.T) {
 	factories.Exporters[typeStr] = factory
 	cfg, err := servicetest.LoadConfigAndValidate(filepath.Join("testdata", "config.yaml"), factories)
 
-	require.NoError(t, err)
+	require.Error(t, err)
 	require.NotNil(t, cfg)
```
[the remark this answers is not in the document]
Comment (anupamdalmia10) [reply exporter/otlphttpexporter/config_test.go:42]: Right, that does look inappropriate.
To incorporate test case for the changes, shall we split up the config.yaml in testdata with one file containing blank otlphttpexporter configuration and the other containing the config as in current config.yaml's otlphttp/2.

[the remark this answers is not in the document]
Comment (bogdandrutu) [reply exporter/otlphttpexporter/config_test.go:42]: I think is ok to have 2 files one ok and one with an error.

**Review by codeboten — COMMENTED** [review]
The settings for otlphttp exporters are validated here:

https://github.com/open-telemetry/opentelemetry-collector/blob/839909863acceff104c18a9db1b5d24c13d6a972/exporter/otlphttpexporter/factory.go#L63-L76

Is the idea to remove that validation as part of this change?

File: exporter/otlphttpexporter/config_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (anupamdalmia10) [reply exporter/otlphttpexporter/config_test.go]: I have split up the config into good and bad config files and updated the test code accordingly.

**Comment by anupamdalmia10** [comment]
@bogdandrutu / @codeboten Is this good to merge?

**Review by bogdandrutu — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**